src/commands/xmentor.ts

```diff
@@ -1,60 +1,89 @@
+      if (!testName) throw new Error('Test name not found');
+      const solutionLanguage = getLanguage(pathToExercise);
+      if (!solutionLanguage) throw new Error('Language not supported');
+      spinner.succeed();
+
+      spinner.start('Installing dependencies.');
+      const installText = await system.run(`npm i`, { cwd: pathToExercise });
+      spinner.succeed(installText);
 
-      spinner.start('Installing dependencies.')
-      const installText = await system.run(`npm i`, { cwd: pathToExercise })
-      spinner.succeed(installText)
+      spinner.start('Unskipping all tests');
+      const fileEnding = LANGUAGE_TEST_FILE_MAPPER[solutionLanguage];
+      const testFile = await filesystem.readAsync(`${pathToExercise}/${testName}.${fileEnding}`);
 
-      spinner.start('Unskipping all tests')
-      const testFile = await filesystem.readAsync(`${pathToExercise}/${fileName}.test.ts`)
-      const testCount = countTests(testFile)
-      const unskippedTests = unskipTests(unskipDescribes(testFile))
+      if (!testFile) throw new Error('Test file not found.');
+      const testCount = countTests(testFile);
+      const unskippedTests = unskipTests(unskipDescribes(testFile));
 
-      await filesystem.writeAsync(`${pathToExercise}//${fileName}.test.ts`, unskippedTests)
-      spinner.succeed(`Found ${testCount} tests`)
+      await filesystem.writeAsync(`${pathToExercise}//${testName}.test.ts`, unskippedTests);
+      spinner.succeed(`Found ${testCount} tests`);
 
-      spinner.start('Running tests.')
-      await system.run(`npx tsc --noEmit -p . && npx jest --i --no-cache`, { cwd: pathToExercise })
-      spinner.succeed('All tests pass')
+      spinner.start('Running tests.');
+      await system.run(LANGUAGE_RUN_COMMAND_MAPPER[solutionLanguage], { cwd: pathToExercise });
+      spinner.succeed('All tests pass');
 
-      const shouldWeDelete = await prompt.confirm('Do you want to delete the student exercise project?')
+      const shouldWeDelete = await prompt.confirm('Do you want to delete the student exercise project?');
 
       if (shouldWeDelete) {
-        spinner.start('Deleting student project')
-        await filesystem.removeAsync(pathToExercise)
-        spinner.succeed()
-        print.success('All done.')
+        spinner.start('Deleting student project');
+        await filesystem.removeAsync(pathToExercise);
+        spinner.succeed();
+        print.success('All done.');
       } else {
-        print.success('All done. Student project is located at \n' + pathToExercise)
+        print.success('All done. Student project is located at \n' + pathToExercise);
       }
     } catch (e) {
-      spinner.fail(e.message)
+      spinner.fail(e.message);
     }
-  }
+  },
+};
+
+export function getTestName(exercisePath: string): string | undefined {
+  const splitPath = exercisePath.split('/');
+  return splitPath[splitPath.length - 1];
+}
+
+export function getLanguage(exercisePath: string): string | undefined {
```

Comment:
Clever, I like what you are doing here, using the dir name to determine the language. We have a problem with the split separator. I'm on a Windows machine and I'm guessing that you are on a Linux machine. When I tried running the program with your changes, I would get a `Language not supported` error and after investigation I have determined that this line is the source of the problem. Here is the value assigned to splitPath on my machine:
`[ 'C:\\Users\\Philippe\\Exercism\\users\\scottbrassfield\\typescript\\leap' ]`

I'm sure you can see the problem. We have the same problem on line 70. Can you try fixing both so that they'll work on Linux as well as Windows.